src/client/input.js

```diff
@@ -10,155 +10,161 @@ import ReactTooltip from 'react-tooltip';
-      delete props.type;
-    }
-
-    const formControl =
-      (children && React.cloneElement(children, props)) ||
-      <FormControl
-        ref={ (c) => { this.refFormControl = c; } }
-        value={ value }
-        { ...props } />;
-
-    const getFormControlWrapped = (className) => (
-       className || hasFeedback || help ?
-        (
-          <div className={ className }>
-            { formControl }
-            { hasFeedback && <FormControl.Feedback /> }
-            { help && <HelpBlock>{help}</HelpBlock> }
-          </div>
-        ) :
-        formControl
-        );
-        
-    if (!addonBefore && !addonAfter && !buttonBefore && !buttonAfter) {
-      return getFormControlWrapped(wrapperClassName);
-    }
-
-    return (
-      <InputGroup
-        bsClass={ cx('input-group', wrapperClassName) }>
-        { this.renderAddon(addonBefore) }
-        { this.renderButton(buttonBefore) }
-        { getFormControlWrapped() }
-        { this.renderButton(buttonAfter) }
-        { this.renderAddon(addonAfter) }
-      </InputGroup>
-    );
-  }
-
-  render() {
-    const {
-      id,
-      label,
-      bsSize,
-      groupClassName,
-      labelClassName,
-      standalone,
-      validationState,
-      tooltipText,
-      ...props
-    } = this.props;
-
-    const helpIconElement = tooltipText && (
+	static propTypes = {
+		name: PropTypes.string,
+		id: PropTypes.string,
+		children: PropTypes.any,
+		help: PropTypes.oneOfType([ PropTypes.string, PropTypes.node ]),
+		bsSize: PropTypes.string,
+		wrapperClassName: PropTypes.string,
+		groupClassName: PropTypes.string,
+		labelClassName: PropTypes.string,
+		addonBefore: PropTypes.any,
+		addonAfter: PropTypes.any,
+		buttonBefore: PropTypes.any,
+		buttonAfter: PropTypes.any,
+		standalone: PropTypes.bool,
+		hasFeedback: PropTypes.bool,
+		validationState: PropTypes.string,
+		label: PropTypes.oneOfType([ PropTypes.string, PropTypes.node ]),
+		type: PropTypes.string,
+		tooltipText: PropTypes.string
+	};
+
+	constructor(props, context) {
+		super(props, context);
+
+		this.refFormControl = null;
+	}
+
+	getDOMNode = () => (
+		ReactDOM.findDOMNode(this.refFormControl)
+	)
+
+	getValue = () => {
+		const inputNode = this.getDOMNode();
+
+		if (this.props.type === 'select' && inputNode.multiple) {
+			return this.getMultipleSelectValues(inputNode);
+		}
+
+		return inputNode.value;
+	}
+
+	getMultipleSelectValues(selectNode) {
+		const values = [];
+		const options = selectNode.options;
+
+		for (let i = 0; i < options.length; i++) {
+			const opt = options[i];
+
+			if (opt.selected) {
+				values.push(opt.value || opt.text);
+			}
+		}
+
+		return values;
+	}
+
+	renderAddon(addon) {
+		return addon && <InputGroup.Addon>{ addon }</InputGroup.Addon>;
+	}
+
+	renderButtons(buttons) {
+		if (Array.isArray(buttons)){
+			return buttons.map((button, index) => this.renderButton(button, index));
+		}
+		return this.renderButton(buttons);
+	}
+	renderButton(button, index=0) {
```

Comment:
Why use a default value for index? That seems unnecessarily risky to me - could result in multiple elements having the same key.

Reply:
I've removed the default index as you suggested, and added a (simple) pagination system, if you want to review that :)